doc/fault_management/fault_mgmt_infra_HLD.md

```diff
@@ -0,0 +1,173 @@
+   - Take fault_type and fault_severity as input from the fetched event and perform lookup
+     on these fields in Fault-Action Policy Table to determine the action(s) needed
+6. Handle the fault (i.e. take action) based on action(s) specified in Fault-Action Policy Table
+   - action may range from logging (disk, OBFL flash etc.) to reload/shutdown etc.
+   - Taking action would either be by itself (i.e. in ts own micro-service) or delegating it to action's owner
+7. Tabulate event entry (along with action taken) for book-keeping purposes
+
+
+# Fault's End-to-End WorkFlow Sequence
+Following workflow depicts the end-to-end fault (event) flow from Fault generation to Fault Handling
+![Fault Management (FM) Workflow sequence](https://github.com/shyam77git/SONiC/assets/69485234/2b453a1b-6e14-48c6-bf61-ab978e62a3bf)
+
+
+# Fault-Action Policy Table (fault_action_policy.json)
+
+{
+
+    "chassis": {
+        "name": "PID or HWSKU",
+        "faults": [
+            {
+        
+                 "type" : "CUSTOM_EVPROFILE_CHANGE",
+                 "severity" : "MAJOR",
+                 "action" : ["syslog"]
+            },
+            {
+                 "type" : "TEMPERATURE_EXCEEDED",
+                 "severity" : "CRITICAL",
+                 "action" : ["syslog", "obfl", "reload"]
+            },
+            {
+                 "type": "FANS MISSING",
+                 "severity": "CRITICAL",
+                 "action" : ["syslog", "obfl", "shutdown"]
+            }
+        ]
+    }
+}
+
+# Use-cases
+Following are some of the Faults' uses-cases 
+| Sr # | Fault Type                    | FDR (Fault source/detector)  |  Fault informant   |  FAH (Fault Analyzer & Handler) |
+|------|-------------------------------|------------------------------|--------------------|---------------------------------|
+| 1.   | Thermal sensors (Temperature) | thermalctld                  |  eventD            |  FM (this HLD/module)           |
+| 2.   | Voltage & Current sensors     | sensormond                   |  eventD            |  FM                             |
+| 3.   | FanTrays and Fans             | thermalctld                  |  eventD            |  FM                             |
+| 4.   | PowerTrays and PSUs           | psud                         |  eventD            |  FM                             |
+| 5.   | transceivers                  | xcvrd                        |  eventD            |  FM                             |
+| 6.   | PCIe faults                   | Host's PCIEd bin/util        |  eventD            |  FM                             |
+| 7.   | Ethernet Switch faults        | platform's ethSwitch service |  eventD            |  FM                             |
+
+# References
+SONiC Events Yang models (schema): 
+- https://github.com/sonic-net/sonic-buildimage/tree/master/src/sonic-yang-models/yang-models
+    - sonic-events-swss.yang
+    - sonic-events-host.yang
+    - sonic-events-bgp.yang etc.
```

Comment:
Are we  planning to take any actions for the events (legacy ones) via fault manager?